File: rx-codelab/src/main/java/com/jraska/rx/codelab/http/GitHubApi.java
package com.jraska.rx.codelab.http;

import java.util.List;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface GitHubApi {
  @GET("/users/{login}")
  Observable<GitHubUser> getUser(@Path("login") String login);

  @GET("/users?since=0")
  Observable<List<GitHubUser>> getFirstUsers();

  @GET("/users/{login}/repos?type=all")
  Observable<List<GitHubRepo>> getRepos(@Path("login") String login);
}

File: rx-codelab/src/main/java/com/jraska/rx/codelab/http/GitHubApi.kt
package com.jraska.rx.codelab.http

import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Path

interface GitHubApi {

  @GET("/users?since=0")
  fun getFirstUsers(): Observable<List<GitHubUser>>

  @GET("/users/{login}")
  fun getUser(@Path("login") login: String): Observable<GitHubUser>

  @GET("/users/{login}/repos")
  fun getRepos(@Path("login") login: String): Observable<List<GitHubRepo>>
}
